ex1_solution/exercise_1_x_mix_drix.py

```diff
@@ -0,0 +1,123 @@
+import copy
+from enum import Enum
+from typing import List, Generator
+
+
+class PlayerToken(Enum):
+    X = 'X'
+    O = 'O'
+    EMPTY = ' '
+
+
+class XOBoard:
+    EMPTY = PlayerToken.EMPTY
+
+    def __init__(self, board: List[List[PlayerToken]]):
+        self._board = copy.deepcopy(board)
+
+    def __str__(self) -> str:
+        dict={PlayerToken.X:"X ",PlayerToken.O:"O ",PlayerToken.EMPTY:"-"}
+        text=""
+        for row in self._board:
+            for i in row:
+                text=text+dict[i]
```

Comment:
Use list comprehension with ''.join(). Something like the following could do the work.
`..join([v.value for v in row])`